### ISSUE TITLE
New key in fiberassign output 

### ISSUE THREAD

**Issue by Srheft** [desihub/fiberassign#209]
@sbailey @tskisner 

Currently:
 To know the number of unassigned **but available** targets in each class (e.g., all the available ELGs that potentially could be assigned but are not), one has to match the TARGETIDs in the first extension ( FIBERASSIGNMENT) with the same column in the 4th extension (POTENTIAL_ASSIGNMENTS) or the target lists in FAVAIL per FIBERID.  

How about:
To remove this **matching step** by adding a keyword *ASSIGN_STATUS* that determines the *assignment status of the target* as opposed to FIBERSTATUS that is more of a hardware status of the positioner/fiber. 

@kdawson1000 if you can/want to add complementary/clarification note to this, please feel free to do so. This is mainly to ease up yours and Christophe+'s effort to assess the completeness status per target class.

**Comment by kdawson1000** [comment]
Yes, we originally thought that the FIBERSTATUS keyword in the "TARGETS" extension was intended to track whether the target was given a fiber.  But I believe that is intended to reflect the status of the hardware, and is not really appropriate for the TARGETS extension.  We could rename it to  ASSIGN_STATUS and use it to track whether a target was given a fiber or not.

**Comment by tskisner** [comment]
Coming back to this open ticket in Spring 2020.  We have had recent discussion about making fiberassign outputs smaller, so adding columns to the large FAVAIL / POTENTIAL_ASSIGNMENTS extensions are probably not what we want.  The `read_assignment_fits_file()` function returns tables from either raw or merged files containing the assignment, targets, and available HDU data.

if fiberassign is run with the option set to write all targets, then the targets table contains all available targets and one can just select rows based on DESI_TARGET values, sum them up, and then see how many are assigned to a fiber.

If fiberassign outputs **do not** include all available targets, then only the FAVAIL / POTENTIAL_ASSIGNMENTS table has information about those available-but-unassigned targets.  Here it only has the TARGETID, so one would have to cross-reference these to the input catalog to get access to the DESI_TARGET values and see how many match your target class of interest.

To summarize:  one can currently write all available targets to the output and this will make larger files that are easy to use to select available targets based on their properties.  Or one can not write all available targets and have smaller files that must be cross-referenced to the input catalogs.
